packages/server-wallet/src/models/adjudicator-status.ts

```diff
@@ -0,0 +1,93 @@
+import {SignedState} from '@statechannels/wallet-core';
+import Knex from 'knex';
+import _ from 'lodash';
+import {Model} from 'objection';
+
+import {Bytes32, Uint48} from '../type-aliases';
+
+export type AdjudicatorStatus =
+  | {
+      status: 'Channel Finalized';
+      finalizedAt: number;
+      states: SignedState[];
+      finalizedBlockNumber: number;
+    }
+  | {status: 'Challenge Active'; finalizesAt: number; states: SignedState[]}
```

Comment:
I'm not sure I understand why we need to store many signed states for these two channel modes. 

In these modes, we store states to enable either 
i) [`pushOutcome`](https://docs.statechannels.org/docs/protocol-tutorial/release-assets#using-pushoutcome) (and friends) or
ii) [`respond`](https://docs.statechannels.org/docs/protocol-tutorial/clear-a-challenge#call-respond)

i) only requires a single `stateHash`, and ii) only requires a single `VariablePart`. So no signatures needed, either. 

Of course there are other methods that require more: such as `checkpoint`. But these are available for the `Open` (or 'Nothing') channel mode, too. In order to `checkpoint` or `challenge`,  I would imagine that the requisite states could be pulled from the existing `channels` table ---because  it has been setup to maintain the support proof for the latest state.

Reply:
I was slightly on the fence about this. I ended up switching the single `state` to `states` to keep it in line with [event](https://github.com/statechannels/statechannels/blob/f2ef253a6bc3b9f7fa85e5ccf2bfaa73358135ad/packages/nitro-protocol/src/contract/challenge.ts#L22) NitroProtocol gives us.

My thinking was that this `AdjudicatorStatus` represents the on-chain status for a channel and I'd rather include more information than less even if it's not actively being used at the moment.

However I don't feel too strongly about it so I'd be happy to switch it back to a single `state`.

Reply:
Good point about the `ChallengeRegistered` event. I'm struggling to remember / understand why it doesn't just emit a single state though. The current test shows us just checking the final state in that array:

https://github.com/statechannels/statechannels/blob/f2ef253a6bc3b9f7fa85e5ccf2bfaa73358135ad/packages/nitro-protocol/test/contracts/ForceMove/challenge.test.ts#L237-L254

@snario do you have some insight about this, given https://github.com/statechannels/statechannels/pull/176 ?